src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php

```diff
@@ -214,7 +214,9 @@ public function regenerate($destroy = false, $lifetime = null)
         }
 
         if (null !== $lifetime) {
```

Comment:
would it make sense to also change this line?
`if (null !== $lifetime && ini_get('session.cookie_lifetime') != $lifetime) {`
that's save saving+starting for nothing, isn't it?

Reply:
indeed.
added the check in d303690442b73562b73d39e581762f1b022278bd